aya/src/programs/tc.rs

```diff
@@ -99,11 +113,43 @@ impl TcAttachType {
             Self::Egress => tc_handler_make(TC_H_CLSACT, TC_H_MIN_EGRESS),
         }
     }
+
+    pub(crate) fn tcx_attach(&self) -> Result<bpf_attach_type, TcError> {
+        match self {
+            Self::Ingress => Ok(BPF_TCX_INGRESS),
+            Self::Egress => Ok(BPF_TCX_EGRESS),
+            Self::Custom(_) => Err(TcError::InvalidTcxAttach),
+        }
+    }
 }
 
-/// Options for SchedClassifier attach
-#[derive(Default)]
-pub struct TcOptions {
+/// Options for a SchedClassifier attach operation. The options vary based on
+/// what is supported by the current kernel. Kernels older than 6.6 must utilize
+/// netlink for attachements, while newer kernels can utilize the modern TCX eBPF
+/// link type which support's the kernel's mprog ordering api.
+pub enum TcAttachOptions {
+    /// Netlink attach options.
+    NlOptions(NlOptions),
+    /// Tcx attach options.
+    TCXOptions(LinkOrder),
```

Comment:
TcAttachOptions (note lowercase c)
TCXOptions (uppercase TCX)
*triggered* :P

I think the variants should be
TcxOrder
Netlink

Unless you think that TcAttachOptions::TcxOptions is going to take more than
just link order?

Reply:
LOL sorry :/, I'm bad a naming

No `TcxOptions` shouldn't ever take more than LinkOrder, fixed based on your recommendations